stable/mongodb-replicaset/templates/mongodb-statefulset.yaml

```diff
@@ -26,6 +26,11 @@ spec:
 {{ toYaml .Values.extraLabels | indent 8 }}
 {{- end }}
       annotations:
+      {{- if .Values.metrics.enabled }}
```

Comment:
You still set the annotations when metrics are enable. You don't need them when using the operator. They should be optional.